Add package test_interface_files

jacobperron opened the pull request:
Connects to https://github.com/ros2/rcl_interfaces/issues/58

This is a refactor of messages from [test_msgs](https://github.com/ros2/rcl_interfaces/tree/master/test_msgs).

I've changed the naming scheme to be more aligned with IDL terminology as discussed in https://github.com/ros2/rosidl/pull/363#issuecomment-484738888.

I've also added the following messages, which are variants of message from [rosidl_generator_c](https://github.com/ros2/rosidl/tree/master/rosidl_generator_c) and [rosidl_generator_cpp](https://github.com/ros2/rosidl/tree/master/rosidl_generator_cpp):
* `BoundedString.msg`
* `PrimitivesConstants.msg`
* `PrimitivesDefault.msg`

I did not include [NestedMessage.action](https://github.com/ros2/rcl_interfaces/blob/1189517a0217a2cb9f78e71c2959542429ad9fc0/test_msgs/action/NestedMessage.action) since it involves testing a nested type from a different package. I think we can leave it in `test_msgs` during the migration.

I've left out many of the other messages defined in [rosidl_generator_c](https://github.com/ros2/rosidl/tree/master/rosidl_generator_c) and [rosidl_generator_cpp](https://github.com/ros2/rosidl/tree/master/rosidl_generator_cpp) because I thought they were redundant or not testing anything in particular. If anyone has a good reason to add them, I'm happy to.

dirk-thomas committed:
move package content into the root of the repo

dirk-thomas committed:
rename package, remove all dependencies, and only install the interface files

jacobperron committed:
Rename messages to align with IDL terminology

Signed-off-by: Jacob Perron <jacob@openrobotics.org>

jacobperron committed:
Update message fields

Update types and field names to match type

Signed-off-by: Jacob Perron <jacob@openrobotics.org>

jacobperron committed:
Add BoundedString.msg, PrimitivesConstants.msg, and PrimitivesDefault.msg

Signed-off-by: Jacob Perron <jacob@openrobotics.org>

jacobperron committed:
Prefix base path to exported lists of interface files

Signed-off-by: Jacob Perron <jacob@openrobotics.org>

dirk-thomas commented on test_interface_files-extras.cmake.in:
```diff
@@ -0,0 +1,31 @@
+# generated from test_interface_files/test_interface_files-extras.cmake.in
+
+set(@PROJECT_NAME@_MSG_FILES "")
+foreach(msg @msg_files@)
```
I would suggest quoting the files (`"@msg_files@"`) to ensure spaces aren't confusing.

(To keep the commit history readable it is probably best to append the change to the commit introducing this.)

Same below.

jacobperron replied:
Done.

jacobperron replied:
hmm.. I think this was the wrong thing to do. Now I only get the prefix added to the first item in the list. Since `msg_files` is a list (e.g.`"A;B;C"`), reverting the change( removing quotes) looks correct. Items in the list with white space seem to be handled properly.

dirk-thomas reviewed (approved):
Besides the single comment this LGTM.

I would suggest to merge without squashing to keep the various in-between steps separate.

jacobperron committed:
Add Various.msg

Move fields for checking alignment of primitives after sequences to Various.msg.

Signed-off-by: Jacob Perron <jacob@openrobotics.org>
